src/network/connection.cpp

```diff
@@ -384,6 +379,48 @@ std::list<BufferedPacket> ReliablePacketBuffer::getTimedOuts(float timeout,
 	return timed_outs;
 }
 
+/*
+	IncomingSplitPacket
+*/
+
+bool IncomingSplitPacket::addChunk(u32 chunk_num, SharedBuffer<u8> chunkdata)
```

Comment:
Copying once in `chunks[chunk_num] = chunkdata;` is enough. Please make it a const ref.

Reply:
`SharedBuffer` is a smart point and will not be implicitly copied